jina/executors/__init__.py

```diff
@@ -241,30 +240,52 @@ def config_abspath(self) -> str:
         """
         return self.get_file_from_workspace(f'{self.name}.yml')
 
+    @staticmethod
+    def get_shard_workspace(workspace_folder: str, workspace_name: str, pea_id: int) -> str:
+        # TODO (Joan, Florian). We would prefer not to keep `pea_id` condition, but afraid many tests rely on this
+        return os.path.join(workspace_folder, f'{workspace_name}-{pea_id}') if pea_id > 0 else os.path.join(workspace_folder)
+
+    @property
+    def workspace_name(self):
+        return self.name
+
+    @property
+    def _workspace(self):
+        """ Property to access `workspace` if existing or default to `./`. Useful to provide good interface when
+        using executors directly in python.
+
+        .. highlight:: python
+        .. code-block:: python
+
+            with NumpyIndexer() as indexer:
+                indexer.touch()
+
+        :return: returns the workspace property of the executor or default to './'
+        """
+        return self.workspace or './'
```

Comment:
I am totally surprised, that this works.
After digging around a little: Seems like I never thought about how boolean expressions work in python. I am totally surprised, that
```
a = 'a' or 'b'
=> a = 'a' (First `True` evaluated element)
a = 'a' and 'b'
=> a = 'b' (Last `True` evaluated element)
a = '' or 'b'
=> a = 'b' (First `True` evaluated element, which is the second one)
a = '' and 'b'
=> a = '' (First `False` evaluated element)
```
When you think how `and` and `or` are implemented (via `__and__` and `__or__`) this makes total sense. I am just amazed... A consequence is:
```
def my_func():
    return 'haha'
if ['a'] and my_func():
    ....
```
breaks down to
```
if 'haha':
    ...
```
which is obviously `True`.